crates/bevy_pbr/src/environment_map/mod.rs

```diff
@@ -0,0 +1,182 @@
+use crate::MeshPipeline;
+use bevy_app::{App, Plugin};
+use bevy_asset::{load_internal_asset, Handle, HandleUntyped};
+use bevy_core_pipeline::prelude::Camera3d;
+use bevy_ecs::{
+    prelude::{Component, Entity},
+    query::With,
+    system::{
+        lifetimeless::{Read, SQuery},
+        Commands, Query, Res, SystemParamItem,
+    },
+};
+use bevy_reflect::{Reflect, TypeUuid};
+use bevy_render::{
+    extract_component::{ExtractComponent, ExtractComponentPlugin},
+    render_asset::RenderAssets,
+    render_phase::{EntityRenderCommand, RenderCommandResult, TrackedRenderPass},
+    render_resource::{
+        BindGroup, BindGroupDescriptor, BindGroupEntry, BindGroupLayout, BindGroupLayoutDescriptor,
+        BindGroupLayoutEntry, BindingResource, BindingType, SamplerBindingType, Shader,
+        ShaderStages, TextureSampleType, TextureViewDimension,
+    },
+    renderer::RenderDevice,
+    texture::Image,
+    RenderApp, RenderStage,
+};
+
+pub const ENVIRONMENT_MAP_SHADER_HANDLE: HandleUntyped =
+    HandleUntyped::weak_from_u64(Shader::TYPE_UUID, 154476556247605696);
+
+pub struct EnvironmentMapPlugin;
+
+impl Plugin for EnvironmentMapPlugin {
+    fn build(&self, app: &mut App) {
+        load_internal_asset!(
+            app,
+            ENVIRONMENT_MAP_SHADER_HANDLE,
+            "environment_map.wgsl",
+            Shader::from_wgsl
+        );
+
+        app.register_type::<EnvironmentMap>()
+            .add_plugin(ExtractComponentPlugin::<EnvironmentMap>::default());
+
+        let Ok(render_app) = app.get_sub_app_mut(RenderApp) else { return };
+
+        render_app.add_system_to_stage(RenderStage::Queue, queue_environment_map_bind_groups);
+    }
+}
+
+/// Environment map based indirect lighting.
+///
+/// When added to a 3D camera, this component adds indirect light
+/// to every point of the scene (including enclosed areas) based on
+/// an environment cubemap texture.
+///
+/// The environment map must be prefiltered into a diffuse and specular map based on the
+/// [split-sum approximation](https://cdn2.unrealengine.com/Resources/files/2013SiggraphPresentationsNotes-26915738.pdf).
+/// The specular map must have exactly 11 mips [0, 10].
+///
+/// To prefilter your environment map, you can use KhronosGroup's [glTF-IBL-Sampler](https://github.com/KhronosGroup/glTF-IBL-Sampler).
+/// The diffuse map uses the Lambertian distribution, and the specular map uses the GGX distribution.
+///
+/// KhronoGroup also has several prefiltered environment maps that can be found [here](https://github.com/KhronosGroup/glTF-Sample-Environments).
```

Comment:
```suggestion
/// To prefilter your environment map, you can use `KhronosGroup`'s [glTF-IBL-Sampler](https://github.com/KhronosGroup/glTF-IBL-Sampler).
/// The diffuse map uses the Lambertian distribution, and the specular map uses the GGX distribution.
///
/// `KhronoGroup` also has several prefiltered environment maps that can be found [here](https://github.com/KhronosGroup/glTF-Sample-Environments).
```

This lint is a bit annoying, but this is why CI is failing